src/SearchEngine.java

```diff
@@ -0,0 +1,111 @@
+import strategies.SearchStrategy;
+import strategies.StrategyALL;
+import strategies.StrategyANY;
+import strategies.StrategyNONE;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+public class SearchEngine {
+
+    Map<String, ArrayList<Integer>> map = new HashMap<>();
+    ArrayList<String> list = new ArrayList<>();
+    static Scanner scanner = new Scanner(System.in);
+
+    public void run(File file) {
+        ArrayList<String> peopleList = readFile(file);
+        Map<String, ArrayList<Integer>> words = mapWords(peopleList);
+        while (true) {
+            int selection = printMenu();
+            switch (selection) {
+                case 1 -> find(peopleList, words);
+                case 2 -> printList(peopleList);
+                case 0 -> {
+                    closeScanner();
+                    System.out.println("\nBye!");
+                    return;
+                }
+                default -> System.out.println("\nIncorrect option! Try again.");
+            }
+        }
+    }
+
+    public ArrayList<String> readFile(File file) {
```

Comment:
`readFile` should live in another place.

Reply:
replaced to FileReader class